### PR TITLE
fix: prevents the application from crashing when a corrupt file is uploaded

### PR DESCRIPTION
**Pull request by faouziH21**
This fixes a bug where when uploading a corrupt (Geo)JSON file the website would crash. Additionally added a new function that checks the viability of the uploaded file. This should enhance the readability of the code.

### CODE REVIEW

File: src/components/ToolMenu/Draw/index.tsx [hunk 1]
```diff
@@ -110,8 +110,9 @@ export const Draw: React.FC<DrawProps> = ({
     }
   };
 
-  const onUploadChange = (e: ChangeEvent<HTMLInputElement>) => {
+  const checkValidityOfUploadFile = (e: ChangeEvent<HTMLInputElement>) => {
     const uploadedFiles = e.target.files;
+    let validitiy = false;
```
Comment (FritzHoing) [inline src/components/ToolMenu/Draw/index.tsx:115]: Is it maybe better to make use of a state here?

**Review by jansule — COMMENTED** [review]
It looks like there will be no feedback if the validity is false, as the only negative feedback is in the catch block

File: src/components/ToolMenu/Draw/index.tsx (hunk 1 above)
Comment (faouziH21) [reply src/components/ToolMenu/Draw/index.tsx:115]: Not sure on how it would be implemented. This keeps the function simple and readable.

Comment (faouziH21) [reply src/components/ToolMenu/Draw/index.tsx:115]: No this method is used in a condition and should not have any side effects.

**Comment by faouziH21** [comment]
> It looks like there will be no feedback if the validity is false, as the only negative feedback is in the catch block

fixed in [e6ab02a](https://github.com/terrestris/shogun-gis-client/pull/1430/commits/e6ab02af598add511898dc219ea055cf9793c9b3)

**Review by dnlkoch — APPROVED** [review]

**Pull request merged**